frontend/apps/reader/modules/readergesture.lua

```diff
@@ -1471,12 +1471,18 @@ function ReaderGesture:gestureAction(action, ges)
     elseif action == "wifi_off" then
         local NetworkMgr = require("ui/network/manager")
         -- can't hurt
-        NetworkMgr:turnOffWifi()
-
-        UIManager:show(InfoMessage:new{
-            text = _("Wi-Fi off."),
-            timeout = 1,
-        })
+        if not NetworkMgr:isOnline() then
```

Comment:
Wifi can be enabled and disconnected at the same time. The "can't hurt" comment means it can't hurt to always try disabling wifi.